src/y0/dsl.py

```diff
@@ -668,8 +668,8 @@ def conditional(self, ranges: VariableHint) -> Expression:
         >>> assert P(A, B).conditional(A) == P(A, B) / Sum[B](P(A, B))
         >>> assert P(A, B, C).conditional([A, B]) == P(A, B, C) / Sum[C](P(A, B, C))
         """
-        ranges = _upgrade_ordering(ranges)
-        ranges_complement = set(self._iter_variables()) - set(ranges)
+        ranges = _upgrade_ordering([r.get_base() for r in ranges])
```

Comment:
I don't agree that the DSL should make these assumptions

Reply:
ID* and IDC* algorithm assume that ranges are only over the base values, not the counterfactual values.

In general,  if I have some joint counterfactual distribution and I want to marginalize out some counterfactual variable, this is how to express it:

```python
P(Y @ -x, Z).marginalize(Y @ -x) == Sum[Y](P(Y @ -x, Z)
```

Reply:
I guess an ambiguity arises if I have this expression:

```python
P(Y @ -x, Y, Z).marginalize(Y@ -x)   
```

Do  I marginalize out `Y @ -z` and `Y`, or just `Y @ -z`?

Reply:
yeah, I get this. maybe for now we can have the algorithm make these decisions before running the marginalize function. Later, we can rework the type hierarchy to make it easier to annotate these constraints, and reinvestigate which versions of this operation make sense at all

Reply:
OR if you're sure that it never makes sense to have a counterfactual variable in the sum's range then we can better document this and it can stay